### PR TITLE
Implement --get-file-link

### PR DESCRIPTION
**Pull request by abraunegg**
* Initial work to implement --get-file-link which will return the weburl of a file which has been synced to OneDrive

### CODE REVIEW

File: src/main.d [hunk 1]
```diff
@@ -371,6 +374,11 @@ int main(string[] args)
 		sync.querySiteCollectionForDriveID(cfg.getValueString("get_o365_drive_id"));
 	}
 	
+	// Are we obtaining the URL path for a synced file?
+	if (cfg.getValueString("get_file_link") != ""){
```
Comment (norbusan) [inline src/main.d]: Maybe add a space between `){`

File: src/sync.d [hunk 2]
```diff
@@ -2635,6 +2635,38 @@ final class SyncEngine
...
+		
+				try {
+					fileDetails = onedrive.getFileDetails(item.driveId, item.id);
+				} catch (OneDriveException e) {
+					log.error("ERROR: Query of OneDrive for file details failed");
+				}
+
+				if ((fileDetails.type() == JSONType.object) && ("webUrl" in fileDetails)) {
+					// Valid JSON object
+					log.log(fileDetails["webUrl"].str);
```
Comment (norbusan) [inline src/sync.d]: Shouldn't that be something like `println`? What if we add an option to turn off logging completely?

File: src/main.d (hunk 1 above)
Comment (abraunegg) [reply src/main.d]: done

File: src/sync.d (hunk 2 above)
Comment (abraunegg) [reply src/sync.d]: done

**Comment by norbusan** [comment]
Thanks!

**Pull request merged**